modules/objc/VectorSeachSnippets.m

```diff
@@ -0,0 +1,120 @@
+//
+//  VectorSeachSnippets.m
+//
+//  Created by Vlad Velicu on 20/03/2024.
+//
+
+#import <Foundation/Foundation.h>
+#import <CouchbaseLite/CouchbaseLite.h>
+#import <NaturalLanguage/NLEmbedding.h>
+
+CBLDatabase* database;
+CBLCollection* collection;
+NLEmbedding* model;
+
+// MARK: Configuring a project to use Vector Search.
+
+/*/
+ Download Couchbase Lite iOS Obj-C and Couchbase Lite Vector Search libraries
+ Extract the xcframeworks and copy them into the App Project Directory
+ 
+ Add both libraries to the *Frameworks, Libraries and Embedded Content* to your desired target
+ */
+
+@interface VectorSearch : NSObject 
+
+@end
+
+@implementation VectorSearch
+
+- (void) createDefaultIndex {
+    NSError* error;
+    database = [[CBLDatabase alloc] initWithName:@"vectorDB" error:&error];
+    collection = [database defaultCollection: &error];
+    
+    // MARK: Create a default Vector Index Configuration
+
+    CBLVectorIndexConfiguration* config = [[CBLVectorIndexConfiguration alloc] initWithExpression: @"vector" dimensions: 300 centroids: 20];
+    
+    // Default values already set:
+    NSLog(@"%@", config.encoding); // .scalarQuantizer(type: .SQ8)
+    NSLog(@"%u", config.metric); // .euclidean
+    NSLog(@"%u", config.minTrainingSize); // 25 * 20
+    NSLog(@"%@", config.encoding); // 256 * 20
+    
+    // MARK: Set custom optional settings
+    config.encoding = [CBLVectorEncoding none];
+    config.metric = kCBLDistanceMetricCosine;
+    config.minTrainingSize = 50;
+    config.maxTrainingSize = 300;
```

Comment:
If the scope of this code snippet is only for creating the default config, these 4 lines are not relevant.

Reply:
those are to show how to set custom settings

Reply:
It is useful to show this, maybe specify with comments that these are custom settings, what they actually change for the end user etc

Reply:
there's line 45 to mark this. I dont think it needs extra code comments within the snippet... plain english documentation should be enough... 

all this information will be available within the api doc as well anyway, so I am not sure if is needed to show "how to assign value to property" at all @pasin

Reply:
@pasin @velicuvlad the current position is good for the purposes of getting the content usable on time for the release. Hakim and I intend to refactor how the mobile docs codebase handles snippets so it's a bit more sustainable in future so there's no need for movement currently